pkg/loki/flow_query.go

```diff
@@ -55,14 +56,39 @@ func NewFlowQueryBuilder(cfg *Config, start, end, limit string, reporter constan
 		}
 	}
 
-	extraLineFilters := []string{}
 	if !utils.Contains(constants.ConnectionTypes, string(recordType)) {
 		if reporter == constants.ReporterSource {
 			labelFilters = append(labelFilters, stringLabelFilter(fields.FlowDirection, "1"))
 		} else if reporter == constants.ReporterDestination {
 			labelFilters = append(labelFilters, stringLabelFilter(fields.FlowDirection, "0"))
 		}
 	}
+
+	extraLineFilters := []string{}
+	if packetLoss != constants.PacketLossAll {
+		sb := strings.Builder{}
+		if packetLoss == constants.PacketLossDropped {
+			// match 0 packet sent and 1+ packets dropped
+			sb.WriteString("|~`")
+			sb.WriteString(`Packets":0[,}]`)
+			sb.WriteString("`")
+			sb.WriteString("|~`")
+			sb.WriteString(`TcpDropPackets":[1-9]*[,}]`)
+			sb.WriteString("`")
+		} else if packetLoss == constants.PacketLossHasDrop {
+			// match 1+ packets dropped
+			sb.WriteString("|~`")
+			sb.WriteString(`TcpDropPackets":[1-9]*[,}]`)
+			sb.WriteString("`")
+		} else if packetLoss == constants.PacketLossSent {
+			// match 1+ packets sent
+			sb.WriteString("|~`")
+			sb.WriteString(`Packets":[1-9]*[,}]`)
```

Comment:
bug spotted: I'm seeing dropped packet in this mode that shouldn't have any, and just figured out why: the regex `Packets":[1-9]*[,}]` matches something like `"TcpDropPackets":0,` 
I guess you should include the leading `"`

Reply:
Another issue: all these regexs won't match packets or dropped with digit 0 in the number, such as "10". They should be instead: `[1-9][0-9]*[,}]`